scripts/compile.sh

```diff
@@ -129,7 +129,7 @@ build_llvm() {
             git fetch --all
         fi
 
-        git checkout dev
+        git checkout main
```

Comment:
let's make it consistent with our CI steps
```suggestion
        IMEX_HASH=$(cat cmake/imex-version.txt)
        git checkout ${IMEX_HASH}
```